### DISCUSSION TITLE
How to use WalletConnect Ethereum Provider?

### DISCUSSION THREAD

**Discussion by orangeqq** [wevm/wagmi#2685] [Q&A]
Hello,

Docs [here](https://wagmi.sh/core/connectors/walletConnect) shows WalletConnectConnector wraps the WalletConnect Ethereum Provider.

But I have no idea how to use it.  I want to get a provider and send it to web3.js.
Could anyone help me, please?

**Comment by olegshilov** [comment]
https://wagmi.sh/react/hooks/useWalletClient

**Reply by orangeqq** [reply]
Thank you for your reply. However, I'm having difficulty understanding the relationship between the [WalletConnect Ethereum Provider] and the [useWalletClient]. Could you please provide more details on this?

Additionally, I'm currently working on obtaining a provider for web3.js in Vanilla JS. I'm referring to the [sample](https://github.com/WalletConnect/web3modal-examples/blob/main/web3modal-wagmi-html-cdn/main.js) that I am in the process of modifying.

Any further insights or guidance you can provide would be greatly appreciated.

**Reply by olegshilov** [reply]
Sorry for the misunderstanding, now I see that you are not using React.
Could you please provide a little more information about what you are trying to achieve, and we can figure it out together)

btw. after you connected you can get provider with this method https://wagmi.sh/core/actions/getWalletClient
Docs about WalletClient is [here](https://viem.sh/docs/clients/wallet.html)

**Reply by orangeqq** [reply]
Thanks. I checked the doc about walletClient, but yet not sure if it is the right thing.
I have a thread in wallet connect discussion [here](https://github.com/orgs/WalletConnect/discussions/2963).
I think a [Ethereum Provider](https://docs.walletconnect.com/2.0/web/providers/ethereum) wrapped with web3modal is necessary for me. But I haven't found how to use it in Vanilla JS.

**Reply by orangeqq** [reply]
Here is my code. It can connect to a MetaMask wallet.
Further more, I want to use web3.js methods to make transactions etc. 
Thus, the provider must be sent to Web3 like this: const web3 = new Web3(provider);

```javascript
import {
  EthereumClient,
  w3mConnectors,
  w3mProvider,
  WagmiCore,
  WagmiCoreChains,
  WagmiCoreConnectors,
} from "https://unpkg.com/@web3modal/ethereum@2.6.2";

import { Web3Modal } from "https://unpkg.com/@web3modal/html@2.6.2";

const { configureChains, createConfig } = WagmiCore;
const { mainnet } = WagmiCoreChains;

const chains = [mainnet];
const projectId = "xxx";

const connector = w3mConnectors({ chains, version: 2, projectId, isNewChainsStale: false });
const w3mprovider = w3mProvider({ projectId });

const { publicClient } = configureChains(chains, [w3mprovider]);
const wagmiConfig = createConfig({
  autoConnect: true,
  connectors: [
    ...connector
  ],
  publicClient,
});

const ethereumClient = new EthereumClient(wagmiConfig, chains);

let web3modal;
const web3 = await getMobileWeb3Instance();
const accounts = web3.eth.getAccounts(); //Since not sending the right provider, I can't use methods of web3.js here

async function getMobileWeb3Instance() {
    try {
      web3modal = new Web3Modal(
        {
          projectId,
          explorerRecommendedWalletIds: [
            'c57ca95b47569778a828d19178114f4db188b89b763c899ba0be274e97267d96',
          ],
          enableExplorer: false,
        },
        ethereumClient
      );
    } catch(e) {
      console.log("Could not get a wallet connection", e);
    }
    const web3 = new Web3(provider); // Here, I want to send a provider to Web3, but have no idea how to get it.
    return web3;
}

```

**Reply by saintmaxi** [reply]
You may need to downgrade to a previous version of web3modal to accomplish this. The latest versions use viem under the hood (with the publicClient convention) rather than ethers which uses provider.

**Reply by orangeqq** [reply]
Thank you, that's a very helpful suggestion!
In the end, I used the session obtained through WalletConnectModal (with Sign API), which can be directly used to send JSON-RPC requests. And that's exactly what I needed.
